sdk/search/search/src/indexDocumentsBatch.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IndexDocumentAction } from "./indexModels";
+
+export class IndexDocumentsBatch<T> {
+  /**
+   * The set of actions taken in this batch.
+   */
+  public readonly actions: IndexDocumentAction<T>[];
+
+  constructor(actions: IndexDocumentAction<T>[] = []) {
+    this.actions = actions;
+  }
+
+  /**
+   * Upload an array of documents to the index.
+   * @param documents The documents to upload.
+   */
+  public upload(documents: T[]): void {
+    const batch = documents.map<IndexDocumentAction<T>>((doc) => {
+      return {
+        actionType: "upload",
+        ...doc
```

Comment:
I didn't realize you were spreading the doc into action.  This is interesting - what's the plan for name conflicts when someone has a field called `actionType`?

Reply:
I mean I could put the actionType after their doc, but not being able to have a key named `actionType` is inherent to the API, isn't it?

Reply:
I don't think so - we send this over the wire as `@search.action`.  I'm assuming AutoRest is transliterating that into `actionType`.

Reply:
Dang, you are right. I should give it a less likely name or leave the original.

Reply:
Renamed to `__actionType`